### DISCUSSION TITLE
Can I create instance of plain c# class via VContainer without registering it?

### DISCUSSION THREAD

**Discussion by Severskiys** [hadashiA/VContainer#661] [Q&A]
For monobehaviour i can use - container.InjectGameObject(gameObject). But if i need plain c# class instance: i register it first (builder.Register<T>()), then resolve (await container and var instance = container.Resolve<T>()).

Can i get similar result with only one operation? For situations when i definitely do not need that instance as dependency for any other class.

**Comment by tigran-sargsyan-w** [comment]
Resolve<T>() itself expects T to already be registered in the container, so there isn't a built-in Resolve<T>() mode that automatically treats every unknown concrete type as transient.

However, if what you want is simply:

> "construct this concrete C# type once and let VContainer resolve/inject its dependencies"

you can build a transient registration on the fly and resolve that registration directly:
```

public static T Create<T>(this IObjectResolver resolver)
{
    var registration =
        new RegistrationBuilder(typeof(T), Lifetime.Transient)
            .Build();

    return (T)resolver.Resolve(registration);
}

```
Then:
`
var instance = container.Create<MyClass>();
`
MyClass itself does not need to be added to the container registry, while constructor/member dependencies used to create it can still be resolved from the existing container.

For objects that are created frequently or are an actual part of your dependency graph, I would still prefer registering a factory:
```

builder.RegisterFactory<MyClass>(
    resolver => () => new MyClass(
        resolver.Resolve<ISomeDependency>()),
    Lifetime.Singleton);

```
But for a truly one-off transient object, resolving an ad-hoc transient Registration is a compact option.

One thing to keep in mind: because this object is outside your normal registered lifetime graph, I would consider the caller responsible for its lifetime/disposal if it owns unmanaged/disposable resources.
